diff --git a/fs/f2fs/super.c b/fs/f2fs/super.c
index a902551dc2741..09e4e5a1ff284 100644
--- a/fs/f2fs/super.c
+++ b/fs/f2fs/super.c
@@ -2221,6 +2221,8 @@ static void default_options(struct f2fs_sb_info *sbi, bool remount)
 	F2FS_OPTION(sbi).memory_mode = MEMORY_MODE_NORMAL;
 	F2FS_OPTION(sbi).errors = MOUNT_ERRORS_CONTINUE;
 
+	set_opt(sbi, ATGC);
+	set_opt(sbi, GC_MERGE);
 	set_opt(sbi, INLINE_XATTR);
 	set_opt(sbi, INLINE_DATA);
 	set_opt(sbi, INLINE_DENTRY);
