diff --git a/fs/f2fs/gc.c b/fs/f2fs/gc.c
index 8b5a55b..3105e35 100644
--- a/fs/f2fs/gc.c
+++ b/fs/f2fs/gc.c
@@ -13,6 +13,8 @@
 #include <linux/delay.h>
 #include <linux/freezer.h>
 #include <linux/sched/signal.h>
+#include <linux/sched.h>
+#include <uapi/linux/sched/types.h>
 #include <linux/random.h>
 #include <linux/sched/mm.h>
 
@@ -227,6 +229,13 @@ int f2fs_start_gc_thread(struct f2fs_sb_info *sbi)
 		return err;
 	}
 
+	/* Run GC kthread in the idle scheduling class */
+	{
+		const struct sched_param param = { .sched_priority = 0 };
+
+		sched_setscheduler(sbi->gc_thread->f2fs_gc_task, SCHED_IDLE, &param);
+	}
+
 	return 0;
 }
 
