diff --git a/lib/decompress_unlz4.c b/lib/decompress_unlz4.c
index e6327391b6b66..57bfe602b8c65 100644
--- a/lib/decompress_unlz4.c
+++ b/lib/decompress_unlz4.c
@@ -162,7 +162,11 @@ STATIC inline int INIT unlz4(u8 *input, long in_len,
 #else
 		dest_len = uncomp_chunksize;
 
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+		ret = LZ4_arm64_decompress_safe(inp, outp, chunksize, dest_len, false);
+#else
 		ret = LZ4_decompress_safe(inp, outp, chunksize, dest_len);
+#endif
 		dest_len = ret;
 #endif
 		if (ret < 0) {
