diff --git a/lib/Kconfig b/lib/Kconfig
index c883045..e269ec2 100644
--- a/lib/Kconfig
+++ b/lib/Kconfig
@@ -652,6 +652,8 @@ source "lib/fonts/Kconfig"
 
 source "lib/intfp/Kconfig"
 
+source "lib/isqrt/Kconfig"
+
 config SG_SPLIT
 	def_bool n
 	help
diff --git a/lib/Makefile b/lib/Makefile
index f873a86..53913f1 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -368,6 +368,8 @@ obj-$(CONFIG_PLDMFW) += pldmfw/
 
 obj-$(CONFIG_INTFP) += intfp/
 
+obj-$(CONFIG_ISQRT) += isqrt/
+
 # KUnit tests
 CFLAGS_bitfield_kunit.o := $(DISABLE_STRUCTLEAK_PLUGIN)
 obj-$(CONFIG_BITFIELD_KUNIT) += bitfield_kunit.o
