diff --git a/lib/decompress_unlz4.c b/lib/decompress_unlz4.c
--- a/lib/decompress_unlz4.c
+++ b/lib/decompress_unlz4.c
@@ -162,7 +162,11 @@
 #else
 		dest_len = uncomp_chunksize;
 
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+		ret = LZ4_arm64_decompress_safe(inp, outp, chunksize, dest_len, false);
+#else
 		ret = LZ4_decompress_safe(inp, outp, chunksize, dest_len);
+#endif
 		dest_len = ret;
 #endif
 		if (ret < 0) {
diff --git a/fs/erofs/decompressor.c b/fs/erofs/decompressor.c
--- a/fs/erofs/decompressor.c
+++ b/fs/erofs/decompressor.c
@@ -238,12 +238,23 @@
 
 	out = dst + rq->pageofs_out;
 	/* legacy format could compress extra data in a pcluster. */
-	if (rq->partial_decoding || !support_0padding)
+	if (rq->partial_decoding || !support_0padding) {
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+		ret = LZ4_arm64_decompress_safe_partial(src + inputmargin, out,
+					  rq->inputsize, rq->outputsize, false);
+#else
 		ret = LZ4_decompress_safe_partial(src + inputmargin, out,
 				rq->inputsize, rq->outputsize, rq->outputsize);
-	else
+#endif
+	} else {
+#if defined(CONFIG_ARM64) && defined(CONFIG_KERNEL_MODE_NEON)
+		ret = LZ4_arm64_decompress_safe(src + inputmargin, out,
+					  rq->inputsize, rq->outputsize, false);
+#else
 		ret = LZ4_decompress_safe(src + inputmargin, out,
 					  rq->inputsize, rq->outputsize);
+#endif
+	}
 
 	if (ret != rq->outputsize) {
 		erofs_err(rq->sb, "failed to decompress %d in[%u, %u] out[%u]",
